include/LightGBM/meta.h

```diff
@@ -25,7 +25,7 @@
 namespace LightGBM {
 
 /*! \brief Type of data size, it is better to use signed type*/
-typedef int32_t data_size_t;
+typedef int64_t data_size_t;
```

Comment:
Add a compile option to switch between int64_t and int32_t.

Reply:
I'll try this, there's a lot of places where there are mixes of refs to `data_size_t` and `int32/64` hopefully, I'll be able to isolate them :)